utils/lib/rr_graph/points.py

```diff
@@ -744,6 +744,285 @@ def straight_closet(line1, line2):
     assert pb != None, "{} {}".format(line1, line2)
     return pa, pb
 
+class Point(object):
+    def __init__(self, coord, tracks=2):
+        self.x, self.y = coord
+        self.tracks = tracks
+
+    def __repr__(self):
+        return 'Point(coord=({},{}),tracks={})'.format(self.x, self.y, repr(self.tracks))
+
+class Track(object):
+    def __init__(self, dim, tracks=None, other_tracks=None, points=[]):
+        self.dim = dim
+
+        self.points = list(points)
+        self.tracks = tracks
+        self.other_tracks = other_tracks
+
+    def __repr__(self):
+        return 'Track(dim={},points={})'.format(repr(self.dim), repr(self.points))
+
+def decompose_points_into_tracks(points):
+    """ This function takes a bag of points and returns a set of x lines and
+        y lines that cover all points, and all lines touch each other.
+
+    This is the first step to forming VPR tracks from points.  VPR tracks have
+    limited length, whereas this function returns lines of infinite length.
```

Comment:
Do you have something which converts the infinite length lines back to points?

Reply:
I'll add something in another PR, but this function is only for generating the list of rows and columns required to cover the points.

Reply:
BTW Does this cover all the things in this diagram?

![image](https://user-images.githubusercontent.com/21212/46386224-09a3c080-c675-11e8-8b74-e3ed363359a3.png)

Reply:
It covers the same test cases that the previous algorithm did.  However, it will generate different tracks.  For example, in the global wire case, it might put the span at the end rather than the middle.  Not sure it matters.

Reply:
I think you do actually want to minimize total path length? (IE Put the spine in the middle) The previous version probably just got it right by accident though.....

Reply:
Why exactly?  Does VPR have channel distance heuristics?

Reply:
In some modes, yes.

Reply:
Okay, if we can about minimizing distances, we need something more general like a minimum spanning tree algo.  I believe https://www.sciencedirect.com/science/article/pii/0166218X9400095U is a suitable algo.  I'll file an issue to see if https://www.sciencedirect.com/science/article/pii/0166218X9400095U is suitable and implement it.